### DISCUSSION TITLE
RabbitMQ Management HTTP API : Field Types

### DISCUSSION THREAD

**Discussion by aslgithub** [rabbitmq/rabbitmq-server#3135] [Questions] [answered]
Hi, 
I'm looking for documentation on the field types used for each field from HTTP API that maybe I cannot obviously find?

The API seems well defined here;
https://rawcdn.githack.com/rabbitmq/rabbitmq-server/v3.8.17/deps/rabbitmq_management/priv/www/api/index.html

However taking the example;
/api/nodes

The field
io_read_avg_time
is defined as 
Average wall time (milliseconds) for each disk read operation in the last statistics interval.

However the raw Json is returning
"io_read_avg_time":0.0,

Which implies the field type is double and not integer {if I am using the right terminology!}
As well as the list of fields is each documented for the data type returned ?

Thanks,
Paul

**Comment by michaelklishin** [comment, marked as answer]
Most metrics are strings and numerical types that should be considered doubles or longs. Until we come around e.g. a Swagger version of the API docs, there isn't a reference but you can inspect what statically typed HTTP API clients such as [Hop](https://github.com/rabbitmq/hop) or [rabbithole](https://github.com/michaelklishin/rabbit-hole) do.
